# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Websevel/FlyRank-Internee/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I choosed freestyle lane beacause i want to work on real-work projects and improve my decision making skills as well as clear all my machine learning concepts in order to get a high quality job in the future.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## 2. Research Question

**Research Question:**
Can machine learning help classify whether water is safe for drinking based on its physical and chemical properties?

**Decision Improved:**
The model will help determine whether a water sample should be considered safe or unsafe.

**Who Acts on It?**
Environmental agencies, water treatment plants, and public health organizations.

**Cost of a Wrong Recommendation:**
If unsafe water is classified as safe, people may consume contaminated water, leading to health risks. If safe water is classified as unsafe, unnecessary treatment and testing costs may increase.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load dataset
df = pd.read_csv("/content_refresh_anonymized.csv")

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

# Missing values
print("\nMissing Values:")
print(df.isnull().sum().sum())

# Target distribution (if Potability exists)
if "Potability" in df.columns:
    print("\nPotability Distribution:")
    print(df["Potability"].value_counts())

Rows: 30000
Columns: 44

Missing Values:
73868


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 3. Quick Look at the Data
 The dataset contain 30,000 rows and 44 columns or features.The data is quite unstructured and irregular due to large number of missing values (73,868).

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.